sdk/index.ts

```diff
@@ -284,53 +315,53 @@ export class MidcontractProtocol {
     }
   }
 
-  async tokenRequireAllowance(owner: Address, amount: number, symbol: SymbolToken = "USDT"): Promise<void> {
-    const allowance = await this.tokenAllowance(owner, symbol);
+  async tokenRequireAllowance(
+    owner: Address,
+    amount: number,
+    symbol: SymbolToken = "USDT",
+    userEscrow: Address
+  ): Promise<void> {
+    const allowance = await this.tokenAllowance(owner, symbol, userEscrow);
     if (allowance < amount) {
       const approveAmount = amount - allowance;
-      const hash = await this.tokenApprove(approveAmount, symbol);
+      const hash = await this.tokenApprove(approveAmount, symbol, userEscrow);
       const transaction = await this.public.waitForTransactionReceipt({ hash });
       if (transaction.status != "success") {
         throw new NotSuccessTransactionError(`token approve of ${symbol} for ${approveAmount}`);
       }
     }
   }
 
-  escrowMakeDataHash(data: string): Promise<Hash> {
-    return this.public.readContract({
-      abi: escrow,
-      address: this.escrow,
-      args: [data],
-      functionName: "makeDataHash",
-    });
-  }
-
-  async getDepositList(depositId: bigint): Promise<Deposit> {
-    const data = await this.public.readContract({
-      address: this.escrow,
-      abi: escrow,
-      functionName: "depositList",
-      args: [depositId],
-    });
-    for (const token of this.tokenList) {
-      if (token.address == data[2]) {
-        return new Deposit([
-          data[0],
-          data[1],
-          token.symbol,
-          Number(formatUnits(data[3], token.decimals)),
-          Number(formatUnits(data[4], token.decimals)),
-          data[5],
-          data[6],
-          data[7],
-          data[8],
-        ]);
-      }
-    }
-    throw new NotFoundError();
-  }
-
-  private async getTransactionReceipt(hash: Hash, waitReceipt = false): Promise<TransactionReceipt | null> {
+  escrowMakeDataHash(data: string): Hash {
+    return ethers.keccak256(ethers.toUtf8Bytes(data)) as Hash;
```

Comment:
Hash calculations can only be performed entirely off-chain if this calculation at the smart contract level is planned to be removed in the future; otherwise it should be done using the contract method

Reply:
https://github.com/midcontract/contracts/blob/main/src/Escrow.sol#L256

Reply:
https://github.com/midcontract/protocol/pull/2/files#diff-66fd8b30e073dfc58786b90e78803df61c90731cd651e24cce5163022f08e43fR330